### PR TITLE
cmd/snap, overlord/snapstate: silently ignore classic flag when a snap is strictly confined

### PR DESCRIPTION
**Pull request by bboozzoo**
Do not fail when attempting to `snap install --classic` a snap that does not require classic confinement. Instead, silently ignore a flag and raise a warning.

This is what it looks like locally:

```
$ snap install test-snapd-tools --classic
test-snapd-tools 1.0 from Canonical✓ installed
Warning: flag --classic ignored for strictly confined snap test-snapd-tools

```

### CODE REVIEW

File: overlord/snapstate/snapstate.go [hunk 1]
```diff
@@ -569,6 +569,10 @@ func InstallPath(st *state.State, si *snap.SideInfo, path, instanceName, channel
 	}
 	info.InstanceKey = instanceKey
 
+	if flags.Classic && !info.NeedsClassic() {
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: this one needs a test, no?

File: overlord/snapstate/snapstate_test.go [hunk 2]
```diff
@@ -1662,6 +1661,37 @@ func (s *snapmgrTestSuite) TestInstallAliasConflict(c *C) {
...
+
+	s.state.Unlock()
+	defer s.se.Stop()
+	s.settle(c)
+	s.state.Lock()
+
+	c.Assert(chg.Err(), IsNil)
+	c.Assert(chg.IsReady(), Equals, true)
+
+	// verify snap is in classic
```
Comment (mvo5) [inline overlord/snapstate/snapstate_test.go]: (nitpick) I guess this should read: `verify snap is *not* in classic mode` (or similar)

File: overlord/snapstate/snapstate.go [hunk 3]
```diff
@@ -632,6 +636,10 @@ func Install(st *state.State, name, channel string, revision snap.Revision, user
 		return nil, err
 	}
 
+	if flags.Classic && !info.NeedsClassic() {
+		// snap does not require classic confinement, silently drop the flag
+		flags.Classic = false
+	}
 	if err := checkInstallPreconditions(st, info, flags, &snapst); err != nil {
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: can we add a TODO about trying to integrate with the helper later (it's probably not trivial for various reasons but slightly appealing)

File: cmd/snap/cmd_snap_op.go [hunk 4]
```diff
@@ -324,6 +324,13 @@ func showDone(cli *client.Client, names []string, op string, esc *escapes) error
 				// TRANSLATORS: the args are a snap name optionally followed by a channel, then a version (e.g. "some-snap (beta) 1.3 installed")
 				fmt.Fprintf(Stdout, i18n.G("%s%s %s installed\n"), snap.Name, channelStr, snap.Version)
 			}
+			if opts != nil && opts.Classic && snap.Confinement != client.ClassicConfinement {
```
Comment (pedronis) [inline cmd/snap/cmd_snap_op.go]: I would put the warning before, not after

**Pull request merged**